architecture/topics/device_manager.adoc

```diff
@@ -0,0 +1,115 @@
+== What is a Device Plugin
+Device Plugin is a gRPC service running on the nodes (external to atomic-openshift-node.service) and is responsible for managing specific HW resources.
+Any Device Plugin must support following RPCs:
+
+[source,golang]
+----
+service DevicePlugin {
+      // ListAndWatch returns a stream of List of Devices
+      // Whenever a Device state change or a Device disappears, ListAndWatch
+      // returns the new list
+      rpc ListAndWatch(Empty) returns (stream ListAndWatchResponse) {}
+
+      // Allocate is called during container creation so that the Device
+      // Plugin can run device specific operations and instruct Kubelet
+      // of the steps to make the Device available in the container
+      rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+}
+----
+
+[[registration]]
+=== Registration
+On start, Device Plugin registers itself with Device Manager invoking `Register` on the `/var/lib/kubelet/device-plugins/kubelet.sock` and starts a gRPC service at `/var/lib/kubelet/device-plugins/<plugin>.sock` for serving Device Manager requests.
+
+[[device-discovery-monitoring]]
+=== Device Discovery and Health Monitoring
+Device Manager, while processing a new registration request, invokes `ListAndWatch` RPC at Device Plugin service. In response, Device Manger gets a list of `Device` objects from Plugin over a `gRPC stream`. Manager will keep watching on the stream for new updates from the Plugin. On the Plugin side, plugin will also keep the stream open and whenever there is a change in the state of any of the devices, a new device list is sent to the Manager over the same streaming connection.
+
+[[device-allocation]]
+=== Device Allocation
+While handling a new pod admission request, Kubelet passes requested `Extended Resources` to the Device Manager for Device allocation. Device Manager checks in its database to verify if a corresponding Plugin exists or not. If Plugin exists and there are free allocatable devices as well as per local cache, `Allocate` RPC is invoked at that particular device plugin.
+
+
+Additionally Device Plugins may also perform several other device specific operations as well such as driver installation, device initialization, device resets etc. These functionalities may vary from implementation to implementation. 
+
+[[example-device-plugin]]
+=== Example Device Plugins
+
+* link:https://github.com/GoogleCloudPlatform/container-engine-accelerators/tree/master/cmd/nvidia_gpu[Nvidia GPU device plugin for COS based OS] 
+* link:https://github.com/NVIDIA/k8s-device-plugin[Nvidia official GPU device plugin]
+* link:https://github.com/vikaschoudhary16/sfc-device-plugin[Solarflare device plugin]
+* link:https://github.com/kubevirt/kubernetes-device-plugins[KubeVirt Device Plugins: vfio and kvm]
+
+TIP: For a simplest Device Plugin reference implementation, there is a stub device plugin in the Device Manager code, `vendor/k8s.io/pkg/kubelet/cm/deviceplugin/stub_device_plugin.go` 
```

Comment:
Should be `vendor/k8s.io/kubernetes/pkg/kubelet/cm/deviceplugin/device_plugin_stub.go`